src/app/AttributePathExpandIterator.h

```diff
@@ -110,10 +128,13 @@ class AttributePathExpandIterator
 
     /// Get the next path of the expansion (if one exists).
     ///
+    /// Can also optionally ask for the corresponding AttributeEntry (e.g. to validate
+    /// read/write options).
```

Comment:
Is this just a convenience so the caller doesn't need to do its own find?

Reply:
Yes, we do a find anyway (we iterate over all attributes) or do a find to determine if the attribute is valid at all. We expect iterators to need to validate things like ACL on things, so may as well return this data. 

Will add comments.

Reply:
> We expect iterators to need to validate things like ACL on things

Why?  This iterator absolutely can't validate ACL, if it's supposed to deal with concrete paths as-is.